### DISCUSSION TITLE
Move_base integration

### DISCUSSION THREAD

**Discussion by PaddyCube** [Fields2Cover/Fields2Cover#85] [Ideas]
Hello and thanks for this great library and especially for documentation of this. In my opinion, even best projects lack on documentation.

To my understanding, the ROS interface actually doesn't support move base yet. I wonder what's the best way to implement it. In my opinion, converting to common nav_msgs/Path should do the trick? Pose might can be calculated in relation to the subsequent point?
I plan to use it on a lawn mower project. The area to cover is given as a polygon with points relative to rtk base station. 

I would implement a ros service which takes polygons as input and returns a path based on your library. 
Would this make sense to you and do you think it makes sense to implement this in your ros package?

BR Patrick

**Comment by Gonzalo-Mier** [comment]
Hi @PaddyCube, welcome to the project!

Actually, there is an issue in the ROS 1 interface asking for it (https://github.com/Fields2Cover/fields2cover_ros/issues/1), but we haven't worked on it. Moreover, @SteveMacenski, @Aposhian and @vinnnyr are  building an out-of-the-box support for Fields2Covern in ROS 2 via Nav2 (https://github.com/Fields2Cover/Fields2Cover/issues/73). Maybe they could answer you about how to work together on the project avoiding overlaps :)

**Comment by vinnnyr** [comment]
although I haven’t worked directly with Move Base in a while, yes, converting to a Path message and publishing it out to Move Base is what you would need to do.

I would make sure everything is done in the correct parent frame. 

When baking in nav2 support, that is essentially what will end up happening

**Reply by PaddyCube** [reply]
Thanks for your answer. So you already solved it for nav2? If so I may can reuse some of your Code. I will use move_base_flex and need to stick to ROS as the robot already use it.

**Comment by vinnnyr** [comment]
There is not much yet to share, it should be just about populating the Path message with each path state from the F2C path planner. and then giving that Path message to MBF or whatever navigation system you want.if you provide a skeleton of what you need I can probably assistOn Aug 18, 2023, at 12:33, PaddyCube ***@***.***> wrote:﻿
Thanks for your answer. So you already solved it for nav2? If so I may can reuse some of your Code. I will use move_base_flex and need to stick to ROS as the robot already use it.
